Scripts/public-api-diff/Tests/PublicApiDiffTests/OutputGeneratorTests.swift

```diff
@@ -0,0 +1,102 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@testable import public_api_diff
+import XCTest
+
+class OutputGeneratorTests: XCTestCase {
+    
+    func test_noChanges_singleModule() {
+        
+        let expectedOutput = """
+        # ✅ No changes detected
+        _Comparing `new_source` to `old_source`_
+
+        ---
+        **Analyzed modules:** Target_1
+        """
+        
+        let outputGenerator = OutputGenerator(
+            changesPerTarget: [:],
+            allTargetNames: ["Target_1"],
+            oldSource: .local(path: "old_source"),
+            newSource: .local(path: "new_source")
+        )
+        
+        let output = outputGenerator.generate()
+        XCTAssertEqual(output, expectedOutput)
+    }
+    
+    func test_oneChange_singleModule() {
+        
+        let expectedOutput = """
+        # 👀 1 public change detected
+        _Comparing `new_source` to `old_source`_
+
+        ---
+        ## `Target_1`
+        - ❇️  Some Addition
+
+        ---
+        **Analyzed modules:** Target_1
+        """
+        
+        let outputGenerator = OutputGenerator(
+            changesPerTarget: ["Target_1": [.init(changeType: .addition, parentName: "", changeDescription: "Some Addition")]],
+            allTargetNames: ["Target_1"],
+            oldSource: .local(path: "old_source"),
+            newSource: .local(path: "new_source")
+        )
+        
+        let output = outputGenerator.generate()
+        XCTAssertEqual(output, expectedOutput)
+    }
+    
+    func test_multipleChanges_multipleModules() {
+        
+        let expectedOutput = """
+        # 👀 7 public changes detected
+        _Comparing `new_source` to `old_repository @ old_branch`_
+
+        ---
+        ## `Target_1`
+        - ❇️  Some Addition
+        - 🔀 Some Change
+        - 😶‍🌫️ Some Removal
+        ## `Target_2`
+        - ❇️  Another Addition
+        - 🔀 Another Change
+        - 😶‍🌫️ Another Removal
+        ### `Parent_In_Target_2`
+        - 🔀 Another Change
+
+        ---
+        **Analyzed modules:** Target_1, Target_2
+        """
+
+        let outputGenerator = OutputGenerator(
```

Comment:
What do you think of `TextOutputGenerator` or `ConsoleOutputGenerator`? Maybe later on you want to add something like `HTMLOutputGenerator`

Reply:
`MarkdownOutputGenerator` would be a good name for the current one - added it to the list of improvements